### ISSUE TITLE
Update() leaks goroutines

### ISSUE THREAD

**Issue by dgryski** [caio/go-tdigest#1]
The AVL iteration loop (spawned from `t.summary.Iter()`) creates a goroutine; terminating the loop early (via break) leaves a goroutune blocked on sending to the current iteration channel.

There is no way I can see with the current AVL library to terminating an iteration early; perhaps switching to a different key/value data structure or implementation would help.

**Comment by dgryski** [comment]
The leaked goroutines is also why `go test -race` fails.

**Comment by caio** [comment]
Aha great, I'll take a look at that ASAP.

Thanks!

**Comment by caio** [comment]
Apparently it still leaks on 1.5

**Comment by caio** [comment]
Build now passes after 19951fc30c0a5d5eda238d3122b5b862e4568524 : https://travis-ci.org/caio/go-tdigest/builds/77023693

TL;DR: It wasn't really leaking
